chips/rp2040/src/pio.rs

```diff
@@ -0,0 +1,1233 @@
+    pub fn init(&self) {
+        let default_config: StateMachineConfiguration = StateMachineConfiguration::default();
+        for state_machine in STATE_MACHINE_NUMBERS {
+            self.sm_config(state_machine, &default_config);
+        }
+    }
+
+    // # Examples
+    // Used for the examples in the pico explorer base main.rs file.
+
+    pub fn blinking_hello_program_init(
+        &mut self,
+        pio_number: PIONumber,
+        sm_number: SMNumber,
+        pin: u32,
+        config: &StateMachineConfiguration,
+    ) {
+        self.sm_config(sm_number, config);
+        self.pio_number = pio_number;
+        self.gpio_init(&RPGpioPin::new(RPGpio::from_u32(pin)));
+        self.sm_set_enabled(sm_number, false);
+        self.set_pins_out(sm_number, pin, 1, true);
+        self.set_set_pins(sm_number, pin, 1);
+        self.sm_init(sm_number);
+        self.sm_set_enabled(sm_number, true);
+    }
+
+    pub fn blink_program_init(
+        &mut self,
+        pio_number: PIONumber,
+        sm_number: SMNumber,
+        pin: u32,
+        config: &StateMachineConfiguration,
+    ) {
+        self.sm_config(sm_number, config);
+        self.pio_number = pio_number;
+        self.gpio_init(&RPGpioPin::new(RPGpio::from_u32(pin)));
+        self.sm_set_enabled(sm_number, false);
+        self.set_pins_out(sm_number, pin, 1, true);
+        self.set_set_pins(sm_number, pin, 1);
+        self.sm_init(sm_number);
+        self.sm_set_enabled(sm_number, true);
+    }
+
+    pub fn sideset_program_init(
+        &mut self,
+        pio_number: PIONumber,
+        sm_number: SMNumber,
+        pin: u32,
+        config: &StateMachineConfiguration,
+    ) {
+        self.sm_config(sm_number, config);
+        self.pio_number = pio_number;
+        self.gpio_init(&RPGpioPin::new(RPGpio::from_u32(pin)));
+        self.gpio_init(&RPGpioPin::new(RPGpio::GPIO7));
+        self.sm_set_enabled(sm_number, false);
+        self.set_pins_out(sm_number, pin, 1, true);
+        self.set_pins_out(sm_number, 7, 1, true);
+        self.set_set_pins(sm_number, pin, 1);
+        self.set_side_set_pins(sm_number, 7);
+        self.sm_init(sm_number);
+        self.sm_set_enabled(sm_number, true);
+    }
+
+    pub fn hello_program_init(
+        &mut self,
+        pio_number: PIONumber,
+        sm_number: SMNumber,
+        pin1: u32,
+        pin2: u32,
+        config: &StateMachineConfiguration,
+    ) {
+        self.sm_config(sm_number, config);
+        self.pio_number = pio_number;
+        self.gpio_init(&RPGpioPin::new(RPGpio::from_u32(pin1)));
+        self.gpio_init(&RPGpioPin::new(RPGpio::from_u32(pin2)));
+        self.sm_set_enabled(sm_number, false);
+        self.set_pins_out(sm_number, pin1, 1, true);
+        self.set_pins_out(sm_number, pin2, 1, true);
+        self.sm_init(sm_number);
+        self.sm_set_enabled(sm_number, true);
+        self.sm_put_blocking(sm_number, 0b11000000);
```

Comment:
could you use a named constant that shows what this binary value means?

Reply:
Sure, switched this to turn_on_gpio_6_7 and added a comment explaining how it works.